xds/src/main/java/io/grpc/xds/RingHashLoadBalancerProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2021 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import io.grpc.Internal;
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancer.Helper;
+import io.grpc.LoadBalancerProvider;
+import io.grpc.NameResolver.ConfigOrError;
+import io.grpc.Status;
+import io.grpc.internal.JsonUtil;
+import io.grpc.xds.RingHashLoadBalancer.RingHashConfig;
+import java.util.Map;
+
+/**
+ * The provider for the "ring_hash" balancing policy.
+ */
+@Internal
+public final class RingHashLoadBalancerProvider extends LoadBalancerProvider {
+
+  @Override
+  public LoadBalancer newLoadBalancer(Helper helper) {
+    return new RingHashLoadBalancer(helper);
+  }
+
+  @Override
+  public boolean isAvailable() {
+    return true;
```

Comment:
This needs to be disabled until we are comfortable with the implementation. Since a service config can list multiple policies, we would want to ignore this policy for now and use any fallbacks until it is deemed stable.

Reply:
We could turn it on by an env variable.

Reply:
To be addressed as the last step, after getting an agreement on the env var name for cross-language.

Reply:
Protected by env var.